modules/puppetca/puppetca_main.rb

```diff
@@ -106,7 +105,7 @@ def find_puppetca
       default_path = ["/opt/puppet/bin", "/opt/puppet/sbin", "/opt/puppetlabs/bin"]
 
       # puppetca is the old method of using puppet cert which is new in puppet 2.6
-      if Puppet::PUPPETVERSION.to_i < 3
+      if Proxy::PuppetCa::Plugin.settings.puppet_version.to_s < '3'
         @puppetca = which("puppetca", default_path) || which("puppet", default_path)
```

Comment:
Yes, that's probably true in most cases.  It might have been written more defensively to begin with (when binary names were switched in 2 to 3) as having multiple versions might have picked the wrong binary.